internal/service/batch/compute_environment.go

```diff
@@ -433,6 +503,55 @@ func resourceComputeEnvironmentUpdate(ctx context.Context, d *schema.ResourceDat
 	return append(diags, resourceComputeEnvironmentRead(ctx, d, meta)...)
 }
 
+func ExpandComputeEnvironmentEC2ConfigurationUpdate(ec2Configuration []interface{}, defaultImageType string) []*batch.Ec2Configuration {
```

Comment:
Note for reviewers: At first glance this code may not appear to be correct based on the AWS documentation.

I spent about 45 minutes on a chime call with an AWS support engineer yesterday running through various update scenarios with the AWS CLI. The code implemented here is based on actual observed behavior. AWS are going to review (and hopefully update) their documentation.